### PR TITLE
[WIP] Azure: Compile with `-g`

### PR DESCRIPTION
**Pull request by ax3l**
Uses default compilation type (`Release`) which implies `-O3`, but adds `-g` debug symbols (just larger binaries) for more useful backtraces.

First seen in #2878.

### CODE REVIEW

**Review by EZoni — APPROVED** [review]
Awesome, looks good to me!

**Comment by ax3l** [comment]
I think with `-g` we get problems with respect to cache size. I checked `du -hs build/` and a single WarpX build with debug symbols `-g` is about 3.2GB in size.

Without `-g`, a default `build/` is just 226 MB. In total, we can only create 10 GB of temporary storage in CI, which also includes all the dependencies and toolchains we need to install...

We can probably optimize a bit by removing `-g` from dependencies:
```console
$ du -hs build/lib/*
1.7M	build/lib/libablastr.a
305M	build/lib/libamrex.a
16K	build/lib/libbuildInfoapp.a
562M	build/lib/libopenPMD.a
```

For our CI, it's probably sufficient to have line numbers on WarpX backtraces and then skip them in AMReX, openPMD-api et al.

File: .azure-pipelines.yml [hunk 1]
```diff
@@ -81,7 +82,7 @@ jobs:
         openmpi-bin libopenmpi-dev \
         libfftw3-dev libfftw3-mpi-dev libhdf5-openmpi-dev pkg-config make \
         python3 python3-pandas python3-pip python3-venv python3-setuptools libblas-dev liblapack-dev
-      ccache --set-config=max_size=10.0G
+      ccache --set-config=max_size=2.0G
```
Comment (ax3l) [inline .azure-pipelines.yml]: Single build with `-g` is 3.2 GB:
```suggestion
      ccache --set-config=max_size=5.0G
```

**Comment by EZoni** [comment]
@ax3l Do you think we should try a `ccache` max. size between 5GB and 10GB here?

**Comment by ax3l** [comment]
I think the issue we face is that the binary caches are not large enough for this.
We could retry this after the CTest transition, maybe we are lucky and are below 10G.

**Comment by EZoni** [comment]
Implemented in #5443.

**Pull request closed without merging**